common/src/test/scala/org/apache/celeborn/common/rpc/RpcSecurityContextSuite.scala

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.celeborn.common.rpc
+
+import org.scalatest.matchers.must.Matchers.be
+import org.scalatest.matchers.should.Matchers.{an, convertToAnyShouldWrapper}
+
+import org.apache.celeborn.CelebornFunSuite
+import org.apache.celeborn.common.exception.CelebornException
+import org.apache.celeborn.common.network.sasl.{SaslCredentials, SecretRegistryImpl}
+import org.apache.celeborn.common.network.sasl.registration.RegistrationInfo
+import org.apache.celeborn.common.security.{ClientSaslContext, ClientSaslContextBuilder, RpcSecurityContextBuilder, ServerSaslContext, ServerSaslContextBuilder}
+
+class RpcSecurityContextSuite extends CelebornFunSuite {
+
+  test("RpcSecurityContext should be created with client and server sasl contexts") {
+    val clientContext = ClientSaslContext(
+      "clientAppId",
+      new SaslCredentials("user", "password"),
+      addRegistrationBootstrap = true)
+    val serverContext = ServerSaslContext(new SecretRegistryImpl())
+
+    val rpcSecurityContext = new RpcSecurityContextBuilder()
+      .withClientSaslContext(clientContext)
+      .withServerSaslContext(serverContext)
+      .build()
+
+    rpcSecurityContext.clientSaslContext shouldBe Some(clientContext)
+    rpcSecurityContext.serverSaslContext shouldBe Some(serverContext)
```

Comment:
Practically, we would be having utmost one in an `RpcSecurityContext` right ? Do we have cases where both are required ?

Reply:
Yes, it will just one. I'll change it to reflect that.